File: FluidStudio/FluidStudioTest/FSProjFileWriterTest.cpp
#include "pch.h"

#include "../FluidStudio/FSProjFileWriter.h"

using namespace PG::FS;

//TEST(FSProjFileWriterTest, TestWrite)
//{
//	FSScene scene;
//	scene.args.maxTimeStep = 0.03;
//
//	FSProjFileWriter writer;
//	const auto actual = writer.write(scene);
//	const double value = actual["MaxTimeStep"];
//	EXPECT_EQ(value, 0.03);
//	//reader.read();
//}

File: FluidStudio/FluidStudioTest/FSProjFileWriterTest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using FluidStudio;
using FluidStudio.FileIO;
using FluidStudio.Physics;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using PG.Core.Math;
using PG.Scene;

namespace FluidStudioTest
{
    [TestClass]
    public class FSProjFileWriterTest
    {
        /*
        [TestMethod]
        public void TestWrite()
        {
            var model = new MainModel();
            var particles = new List<Vector3d>();
            particles.Add(new Vector3d(0, 0, 0));
            var world = new SceneList();
            var id = world.AddParticleSystemScene(particles, "", new PG.Core.UI.ParticleAppearance(), 1);
            var physicsScene = new SolverScene();
            var fluidScene = new FluidScene(model.Scenes, id, 1.0f, 1.0f, "Fluid01");
            physicsScene.Fluids.Add(fluidScene);
            model.PhysicsModel.Scenes.Add(physicsScene);
            var writer = new FSProjFileWriter();
            writer.Write(model, "TestWrite.fsproj");
        }
        */
    }
}
